x-pack/plugins/cases/server/files/index.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FileJSON, FileKind } from '@kbn/files-plugin/common';
+import type { FilesSetup } from '@kbn/files-plugin/server';
+import {
+  APP_ID,
+  constructFilesHttpOperationTag,
+  MAX_FILE_SIZE,
+  OBSERVABILITY_OWNER,
+  SECURITY_SOLUTION_OWNER,
+} from '../../common/constants';
+import type { Owner } from '../../common/constants/types';
+import { HttpApiTagOperation } from '../../common/constants/types';
+import { ALLOWED_MIME_TYPES, IMAGE_MIME_TYPES } from '../../common/constants/mime_types';
+
+const buildFileKind = (owner: Owner): FileKind => {
+  return {
+    id: owner,
+    http: fileKindHttpTags(owner),
+    maxSizeBytes,
+    allowedMimeTypes: ALLOWED_MIME_TYPES,
+  };
+};
+
+const fileKindHttpTags = (owner: Owner) => {
+  return {
+    create: buildTag(owner, HttpApiTagOperation.Create),
+    delete: buildTag(owner, HttpApiTagOperation.Delete),
+    download: buildTag(owner, HttpApiTagOperation.Read),
+    getById: buildTag(owner, HttpApiTagOperation.Read),
+    list: buildTag(owner, HttpApiTagOperation.Read),
+    update: buildTag(owner, HttpApiTagOperation.Update),
+  };
+};
+
+const access = 'access:';
+
+const buildTag = (owner: Owner, operation: HttpApiTagOperation) => {
+  return {
+    tags: [`${access}${constructFilesHttpOperationTag(owner, operation)}`],
+  };
+};
+
+const MAX_IMAGE_FILE_SIZE = 10 * 1024 * 1024; // 10 MiB
```

Comment:
nit: As it is a constant maybe we should put it on `x-pack/plugins/cases/common/constants/files.ts`

Reply:
Yeah, I think my thought was, should we keep putting constants that only the server side needs in `common`? My understanding of `common` was that it should be for stuff that either an external plugin would use or that the frontend and backend need. What do you think?

Reply:
You are right! Let's keep it as it is for the moment. I forgot that the UI does not use it.